### PR TITLE
Created endpoints for fetching and deleting users as superadmin

### PR DESCRIPTION
**Pull request by AlexandraVoloshyna**
## Describe your changes

- implemented get and delete endpoints 
- implemented filters and pagination
- added seeder to populate database users table

## Issue ticket code (and/or) and link

- [Link to JIRA ticket](https://olga-team.atlassian.net/browse/TP-119)

### **General**

- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [x] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [x] Don't have "any" on my code
- [x] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside the ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file.
- [x] Used camelCase for variables and functions
- [ ] Date and time formats are on the constants
- [x] Functions are public only if it's used outside the class
- [x] No hardcoded values
- [ ] covered by tests
- [x] Check your commit messages meet the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/)

### Backend

- [x] Swagger documentation updated
- [x] Database requests are optimized and not redundant
- [ ] Unit tests written
- [x] use ConfigService instead of process.env
- [ ] use transactions if there is a call chain that mutates data in different tables
- [ ] use @index decorator for frequently requested data

### CODE REVIEW

File: src/modules/user/dto/filter-users.dto.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+export class UsersFilterDto {
+  @ApiProperty()
+  @IsOptional()
+  @IsString()
+  search: string;
+
+  @ApiProperty()
+  @IsOptional()
+  @IsString()
+  order?: 'DESC' | 'ASC';
```
Comment (MishaFomenko) [inline src/modules/user/dto/filter-users.dto.ts]: maybe put this into enum

File: src/modules/user/services/user-seed.service.ts [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+        phone: '8899001122',
+        address: '707 Ash St',
+        address_2: null,
+        country: 'China',
+        city: 'Beijing',
+        cloth_size: 'L',
+        jeans_size: '34',
+        shoes_size: 9,
+      },
+    ];
```
Comment (MishaFomenko) [inline src/modules/user/services/user-seed.service.ts]: maybe move these to constants

File: src/modules/user/services/user.service.ts [hunk 3]
```diff
@@ -62,6 +65,37 @@ export class UserService {
...
+      where: conditions,
+      order: { full_name: order },
+      skip: (page - 1) * limit,
+      take: limit,
+    });
+    return { totalCount, users };
+  }
+
+  public async softDelete(id: string): Promise<void> {
+    const user = await this.isUserExist(id);
```
Comment (MishaFomenko) [inline src/modules/user/services/user.service.ts:100]: need to check if user was found

File: src/modules/user/services/user.service.ts [hunk 4]
```diff
@@ -62,6 +65,37 @@ export class UserService {
...
     }
 
-    await this.userRepository.update(id, dtoWithoutPassword);
+    await this.userRepository.update(id, {
+      password: user.password,
+      ...dtoWithoutPassword,
+    });
+  }
+
+  public async getAllUsers(dto: UsersFilterDto) {
```
Comment (MishaFomenko) [inline src/modules/user/services/user.service.ts]: output type

File: src/modules/user/services/user.service.ts (hunk 3 above)
Comment (AlexandraVoloshyna) [reply src/modules/user/services/user.service.ts:100]: isUserExist method has this check already

**Pull request merged**